core/templates/components/shared-component.module.ts

```diff
@@ -21,16 +21,30 @@ import 'zone.js';
 import { NgModule } from '@angular/core';
 import { CommonModule } from '@angular/common';
 
+import { FlexLayoutModule } from '@angular/flex-layout';
+
 import { BackgroundBannerComponent } from
   './common-layout-directives/common-elements/background-banner.component';
 import { TranslatePipe } from 'filters/translate.pipe';
+import { SharingLinksComponent } from
+  './common-layout-directives/common-elements/sharing-links.component';
+import { ExplorationEmbedButtonModalComponent } from
+  './button-directives/exploration-embed-button-modal.component';
+import { NgbModalModule } from '@ng-bootstrap/ng-bootstrap';
 
 
 @NgModule({
-  imports: [CommonModule],
-  declarations: [BackgroundBannerComponent, TranslatePipe],
-  entryComponents: [BackgroundBannerComponent],
-  exports: [BackgroundBannerComponent, TranslatePipe]
+  imports: [CommonModule, FlexLayoutModule, NgbModalModule],
+  declarations: [BackgroundBannerComponent, TranslatePipe,
```

Comment:
How do we maintain these arrays? Should this be sorted? If we keep it sorted, adding a directive will require a change from the top to the bottom, this would be hard.

Reply:
Let's not keep them sorted, for now, we can just add them to the end. Once we are done with migration we can sort them at once, always felt that adding something to an array was really cumbersome because it affected the lines below it too.